Init notifications in the correct location

jimmynotjim opened the pull request:
Currently notifications are initialized within the formSubmit which lead
to errors where the notifications weren't being updated correctly (a new 
instance was created every submission). Moving them to their direct 
parent and passing them into the formSubmit resolves the issue.

## Changes

- Moves notification init to it's parent function
- Passes notification into the formSubmit function

## Testing

1. Run `gulp test:unit` and everything should continue to pass

## Checklist

- [x] PR has an informative and human-readable title
- [x] Changes are limited to a single goal (no scope creep)
- [x] Code can be automatically merged (no conflicts)
- [x] Code follows the standards laid out in the [development playbook](https://github.com/cfpb/development)
- [x] Passes all existing automated tests
- [x] Any _change_ in functionality is tested
- [x] New functions are documented (with a description, list of inputs, and expected output)
- [x] Placeholder code is flagged / future todos are captured in comments
- [x] Visually tested in supported browsers and devices (see checklist below :point_down:)
- [x] Project documentation has been updated
- [x] Reviewers requested with the [Reviewers tool](https://help.github.com/articles/requesting-a-pull-request-review/) :arrow_right:

anselmbradford commented:
Can you explain in more detail the circumstances where this was arising? Like how would I reproduce the issue on the master branch?

jimmynotjim commented:
@anselmbradford the issue came up during the SVG Icon transition and I regret not pulling it out of that PR sooner. It's been so long I've forgotten the exact details

Essentially the issue was that the notification exists in the static markup and the init searches it out relative to the passed HTMLNode. When the forms were submitted, it was re-initializing on every submission, rather than updating the existing instance. This was causing issues where the notification couldn't update the existing icon https://github.com/cfpb/cfgov-refresh/pull/3887/files#diff-32e184401b65625fc114c891e29b833cR130

jimmynotjim commented:
Turns out the issue was slightly different than we originally thought. The notification was never initialized within the formSubmit and when we moved the instance to the files that call the formSubmit we fixed it just by initializing the notification. Updated the formSubmit to initialize (as well as create the instance) within it's own init method and everything is working as expected again.

anselmbradford commented on test/unit_tests/js/organisms/FormSubmit-spec.js:
```diff
@@ -0,0 +1,69 @@
+      </div>
+
+      <div class="form-group">
+        <input type="hidden" name="code" value="USCFPB_91">
```
Do we need the value code here in the test?

jimmynotjim replied:
I don't think so, removed and amended the commit.

anselmbradford commented:
LGTM +1

You could stack the attributes to wrap the long lines in the test HTML_SNIPPET if you wanted.

jimmynotjim commented:
Good call, fixed the attributes in the fixture.
